src/boost_histogram/_internal/hist.py

```diff
@@ -98,11 +98,11 @@ def __init__(self, *axes, **kwargs):
 
         # Keyword only trick (change when Python2 is dropped)
         with KWArgs(kwargs) as k:
-            storage = k.optional("storage", Double())
+            self._storage = k.optional("storage", Double())
```

Comment:
You can access the storage type from the C++ object, which is better than trying to hang on to it in Python.